lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -255,6 +255,7 @@ class AzurePlatformSchema:
     availability_set_properties: Optional[Dict[str, Any]] = field(default=None)
     vm_tags: Optional[Dict[str, Any]] = field(default=None)
     locations: Optional[Union[str, List[str]]] = field(default=None)
+    use_private_ip: bool = field(default=False)
```

Comment:
It can be a flag on RemoteNode in `schema.py`, so it can be handled in set_connection_info of node.py.

Reply:
hmmm ... it seems i don't understand what all is needed to use a flag on RemoteNode instead of what i've done here.  can you provide more details?

Reply:
would the RemoteNode flag be used instead of this AzurePlatformSchema flag?  or in addition to?

Reply:
updated the PR with my best guess of what you had in mind :).  let me know what i got wrong ... once i have this correct, i'll rerun my validation to make sure it still works.

Reply:
Thank you, that most like what I thought!